instrumentation-core/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/ExporterAutoConfiguration.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.exporters;
+
+import io.opentelemetry.exporters.logging.LoggingSpanExporter;
+import io.opentelemetry.instrumentation.spring.autoconfigure.exporters.jaeger.JaegerExporterFactory;
+import io.opentelemetry.instrumentation.spring.autoconfigure.exporters.jaeger.JaegerExporterProperties;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties({ExporterProperties.class, JaegerExporterProperties.class})
+@ConditionalOnProperty(prefix = "opentelemetry.exporter", name = "enabled", matchIfMissing = true)
+public class ExporterAutoConfiguration {
+
+  @Bean
+  @Autowired
+  @ConditionalOnMissingBean
+  public SpanExporter otelSpanExporter(
+      ExporterProperties exporterProperties, JaegerExporterProperties jaegerExporterProperties) {
```

Comment:
This isn't clear to me, perhaps because I'm misunderstanding the dependency injection. 
Would that require Jaeger properties for the method to run, or can they be @Nullable, or Optional?
Otherwise, I'd consider having another method that is invoked only in the Jaeger case.

Also, do you need some kind of precondition check to make sure the properties are actually available in the case a user has chosen jaeger?

Reply:
I changed this implementation but each exporter is optional. Users must set the `opentelemetry.trace.exporter.jaeger.enabled=true` in their application.properties file to send traces to jaeger. Otherwise this exporter will not be configured. The only exporter that is configured by default is the simple LoggingSpanExporter but this can be turn off by setting `opentelemetry.trace.exporter.logging.enabled=false`